dist/knockout.validation.d.ts

```diff
@@ -0,0 +1,446 @@
+            /** The rule condition. */
+            onlyIf?: () => boolean;
+        }
+
+        export interface ValidationAsyncRuleDefinition {
+            /** Indicates that this rule is async. */
+            async: true;
+            /** The rule message. */
+            message: string;
+            /** The rule async validator. */
+            validator(value: any, params: any, callback: ValidationAsyncCallback): void;
+            /** The rule condition. */
+            condition?: () => boolean;
+            /** The rule condition. */
+            onlyIf?: () => boolean;
+        }
+
+        export interface ValidationGroup extends ValidationGroupFunctions {
+            (): string[];
+        }
+
+        export interface ValidationGroupComputed extends ko.Computed<string[]>, ValidationGroupFunctions { }
+
+        export interface ValidationGroupFunctions {
+            showAllMessages(show?: boolean): void;
+            isAnyMessageShown(): boolean;
+            filter(predicate: (obs: ko.Observable<any> | ko.ObservableArray<any> | ko.Computed<any>) => boolean): Array<ko.Observable<any> | ko.ObservableArray<any> | ko.Computed<any>>;
+            find(predicate: (obs: ko.Observable<any> | ko.ObservableArray<any> | ko.Computed<any>) => boolean): ko.Observable<any> | ko.ObservableArray<any> | ko.Computed<any>;
+            forEach(predicate: (obs: ko.Observable<any> | ko.ObservableArray<any> | ko.Computed<any>) => void): void;
+            map<U>(predicate: (obs: ko.Observable<any> | ko.ObservableArray<any> | ko.Computed<any>) => U): U[];
+        }
+
+        export type ObjectValidationRuleDefinitions<T = any> = {
+            [P in keyof T]:
+            T[P] extends object ? ObjectValidationRuleDefinitions<T[P]> :
+            T[P] extends Array<infer U> ? ObjectValidationRuleDefinitions<U> :
+            { [key: string]: ValidationRuleExtenderParams | ValidationAnonymousRuleDefinition; }
+        }
+
+        export type ValidationAsyncCallback = (result: boolean | { isValid: boolean; message: string; }) => void;
+
+        export type ValidationAnonymousRuleDefinition = ValidationRuleDefinition | ValidationAsyncRuleDefinition;
+
+        export interface ValidationExtendOptions {
+            validation?: ValidationRule | ValidationRuleDefinition | ValidationAsyncRuleDefinition | Array<ValidationRule | ValidationRuleDefinition | ValidationAsyncRuleDefinition>;
+            validatable?: boolean | ExtenderValidatableOptions;
+
+            required?: boolean | ValidationRuleExtenderParams<boolean>;
+            min?: number | ValidationRuleExtenderParams<number>;
+            max?: number | ValidationRuleExtenderParams<number>;
+            minLength?: number | ValidationRuleExtenderParams<number>;
+            maxLength?: number | ValidationRuleExtenderParams<number>;
+            pattern?: RegExp | ValidationRuleExtenderParams<RegExp>;
+            step?: RegExp | ValidationRuleExtenderParams<RegExp>;
```

Comment:
the step?:  line should match min?: rather than pattern?:
this line should be:
step?: number | ValidationRuleExtenderParams<number>

Reply:
Done!